fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2075,12 +2232,15 @@ public Range getUnbuiltRange() {
                                                                                   @Nonnull IsolationLevel indexStateIsolationLevel) {
         CompletableFuture<RecordMetaDataProto.DataStoreInfo> storeHeaderFuture = loadStoreHeaderAsync(existenceCheck, storeHeaderIsolationLevel);
         CompletableFuture<Map<String, IndexState>> loadIndexStates = loadIndexStatesAsync(indexStateIsolationLevel);
-        return context.instrument(FDBStoreTimer.Events.LOAD_RECORD_STORE_STATE, storeHeaderFuture.thenCombine(loadIndexStates, RecordStoreState::new));
+        CompletableFuture<Map<String, IndexMetaDataProto.IndexMetaData>> loadIndexMetaData = loadIndexMetaDataAsync(indexStateIsolationLevel);
```

Comment:
I think we should probably consider whether or not the metadata can be lazily loaded.  Here we are going to pay the penalty for the reading of the metadata pretty much every time the store is opened, yet the metadata is not really needed during normal "healthy" store operation.